dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.core.models.Region;
+import uk.ac.ebi.eva.dbsnpimporter.models.SubSnpCoreFields;
+import uk.ac.ebi.eva.dbsnpimporter.sequence.SequenceReader;
+
+import java.util.NoSuchElementException;
+
+public class AssemblyCheckFilterProcessor implements ItemProcessor<SubSnpCoreFields, SubSnpCoreFields> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AssemblyCheckFilterProcessor.class);
+
+    private SequenceReader assemblyReader;
+
+    public AssemblyCheckFilterProcessor(SequenceReader assemblyReader) {
+        this.assemblyReader = assemblyReader;
+    }
+
+    @Override
+    public SubSnpCoreFields process(SubSnpCoreFields subSnpCoreFields) throws Exception {
+        String referenceAllele = subSnpCoreFields.getReferenceInForwardStrand();
+        // check if reference is not empty
+        if (referenceIsNotEmpty(referenceAllele)) {
+            Region region = subSnpCoreFields.getVariantCoordinates();
+            try {
+                String sequenceInAssembly = assemblyReader.getSequence(region.getChromosome(), region.getStart(),
+                                                                       region.getEnd());
+                if (referenceAllele.equals(sequenceInAssembly)) {
+                    return subSnpCoreFields;
+                } else {
+                    logger.warn("Assembly check failed for variant {}. {} expected in {}, {} found",
+                                subSnpCoreFields.getSsId(), sequenceInAssembly, region, referenceAllele);
+                    return null;
+                }
+            } catch (IndexOutOfBoundsException  | NoSuchElementException | IllegalArgumentException e) {
+                logger.error("Incorrect variant region {}: {}", region, e.getMessage());
```

Comment:
Under what conditions will this happen? Should this error abort the application?

Reply:
If the variant chromosome is not in the fasta file, or the coordinates are wrong (end < start), or the coordinates are greater than the number of nucleotides that has the chromosome. I don't expect to see many of those errors,  so I'm not stopping the execution, but I wanted to take a further look to those variants if that happens and for that reason, I'm logging those at ERROR level. I will update the log message according to the other processors style.

Reply:
This is not an ERROR message anymore. Now I' m not catching the different exceptions separate, just logging the error messages would be enough to detect if something is happening.